### PR TITLE
NightVision - Fix NVS blur

### LINKED ISSUE #6666: NVS scope has blur
**Arma 3 Version:** `1.84` (stable)
**CBA Version:** `3.9.0` (stable)
**ACE3 Version:** `3.12.3` (stable / dev + commit hash)

**Mods:**
```
- CBA_A3
- ace
```

**Description:**
- When you aim down the NVS (vanilla scope with integrated night vision) the screen is blurred like when you aim down any normal sight with NVG, however when looking in binoculars with night vision (like the Vector nite) the screen is not blurred; ithink NVS shouldn't be blurred.

Vector : https://i.imgur.com/DPakw0u.jpg
NVS : https://i.imgur.com/QNccjJ2.jpg

**Steps to reproduce:**
- Place a unit, give him the NVS, aim down the sight.

**Where did the issue occur?**
- Dedicated / Editor (Singleplayer)

**Placed Modules:**
- 

**RPT log file:**
-

### PR DESCRIPTION
**Pull request by kamkill01011**
**When merged this pull request will:**
- Fix #6666 by removing blur in NVS.

### CODE REVIEW

**Comment by commy2** [comment]
Hard coded classnames are a terrible idea. 👎

**Comment by commy2** [comment]
`linearConversion` from 1 to 1?

**Comment by kamkill01011** [comment]
> `linearConversion` from 1 to 1?

I don't understand what you are saying.

File: addons/nightvision/functions/fnc_pfeh.sqf [hunk 1]
```diff
@@ -85,6 +85,10 @@ if (CBA_missionTime < GVAR(nextEffectsUpdate)) then {
     // Modify blur if looking down scope
     if ((cameraView == "GUNNER") && {[ACE_player] call CBA_fnc_canUseWeapon}) then {
         if (currentWeapon ACE_player == "") exitWith {};
+        if (
+            currentWeapon ACE_player == primaryWeapon ACE_player && 
+            count ("'NVG' in getArray (_x >> 'visionMode')" configClasses (configFile >> "CfgWeapons" >> ((primaryWeaponItems ACE_player) select 2) >> "ItemInfo" >> "OpticsModes")) > 0
+        ) exitWith {_blurFinal = _blurFinal * linearConversion [0, 1, GVAR(aimDownSightsBlur), 1, ST_NVG_CAMERA_BLUR_SIGHTS_NVS]}; // NVS are good
```
Comment (commy2) [inline addons/nightvision/functions/fnc_pfeh.sqf]:
```sqf
#define ST_NVG_CAMERA_BLUR_SIGHTS_NVS 1
linearConversion [0, 1, GVAR(aimDownSightsBlur), 1, ST_NVG_CAMERA_BLUR_SIGHTS_NVS]
```
`linearConversion` from 1 to 1 makes no sense.

Comment (kamkill01011) [reply addons/nightvision/functions/fnc_pfeh.sqf]: Yes absolutely, I think there is no need to update the `_blurFinal` value.

**Comment by commy2** [comment]
Doesn't work if I put the optic on a pistol or launcher.
Does change effect for CQB optics, e.g. the BWA3 ZO4x30 with NVD.

**Comment by PabstMirror** [comment]
I couldn't figure out a good way to tell if using NVG or the co-optic

Now I wonder if we could use something like
`zoom = "isUsingNVG = true; 0.25";`
in the optic configs?

**Comment by commy2** [comment]
@PabstMirror 
In the new CBA optics component, I use a hard coded idc control in the weapon info type display.
https://github.com/CBATeam/CBA_A3/blob/master/addons/optics/RscInGameUI.hpp#L16-L18
https://github.com/CBATeam/CBA_A3/blob/master/addons/optics/fnc_animateScriptedOptic.sqf#L37-L38

This control is hidden in CQB mode.

File: addons/nightvision/functions/fnc_pfeh.sqf [hunk 2]
```diff
@@ -85,6 +85,7 @@ if (CBA_missionTime < GVAR(nextEffectsUpdate)) then {
     // Modify blur if looking down scope
     if ((cameraView == "GUNNER") && {[ACE_player] call CBA_fnc_canUseWeapon}) then {
         if (currentWeapon ACE_player == "") exitWith {};
+        if (count ("'NVG' in getArray (_x >> 'visionMode')" configClasses (configFile >> "CfgWeapons" >> ((ACE_player weaponAccessories currentWeapon ACE_player) select 2) >> "ItemInfo" >> "OpticsModes")) > 0) exitWith {}; // NVS are good
```
Comment (PabstMirror) [inline addons/nightvision/functions/fnc_pfeh.sqf]: switch to findIf (will do)

Comment (PabstMirror) [reply addons/nightvision/functions/fnc_pfeh.sqf]: look at inheritance/configProperties

**Comment by commy2** [comment]
This PR has many issues and does not fix the problem, but I am too lazy to point them out. Instead, replace this with a proper fix in the PR linked above.

**Pull request closed without merging**